### PR TITLE
Testjob page

### PR DESCRIPTION
**Pull request by mwasilew**

### CODE REVIEW

File: squad/ci/backend/lava.py [hunk 1]
```diff
@@ -90,6 +90,15 @@ def listen(self):
                     # TODO: at least log error
                     pass
 
+    def job_url(self, job_id):
```
Comment (terceiro) [inline squad/ci/backend/lava.py]: the other methods take a job, I think it would be good to maintain consistencty

File: squad/ci/backend/null.py [hunk 2]
```diff
@@ -44,3 +44,9 @@ def listen(self):
         the received data is up to each specific backend implementation.
         """
         pass
+
+    def job_url(selt, job_id):
```
Comment (terceiro) [inline squad/ci/backend/null.py]: ditto

File: squad/frontend/templates/404.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{% extends "squad/base.html" %}
+{% load squad %}
+
+{% block content %}
+<h1>Page not found!<h1>
+{% endblock %}
```
Comment (terceiro) [inline squad/frontend/templates/404.html]: this is not used

File: squad/frontend/views.py [hunk 4]
```diff
@@ -185,3 +186,16 @@ def metrics(request, group_slug, project_slug):
         "data": data,
     }
     return render(request, 'squad/metrics.html', context)
+
+
+def test_job(request, testjob_id):
+    testjob = get_object_or_404(TestJob, pk=testjob_id)
+    if testjob.job_id is not None and testjob.url is not None:
```
Comment (terceiro) [inline squad/frontend/views.py]: the first test duplicates test that is already done by `Testjob.url`

File: squad/frontend/templates/404.html (hunk 3 above)
Comment (mwasilew) [inline squad/frontend/templates/404.html]: It's used in this line:
testjob = get_object_or_404(TestJob, pk=testjob_id)
In general this template will be used for any 404 that might happen in squad. Do we have any other 404 template I'm not aware of?

File: squad/ci/backend/lava.py (hunk 1 above)
Comment (mwasilew) [inline squad/ci/backend/lava.py]: Fixed

File: squad/ci/backend/null.py (hunk 2 above)
Comment (mwasilew) [inline squad/ci/backend/null.py]: Fixed

**Comment by mwasilew** [comment]
This failure looks strange. I'm not able to reproduce it locally:
https://travis-ci.org/mwasilew/squad/jobs/231207449

File: squad/frontend/templates/404.html (hunk 3 above)
Comment (terceiro) [inline squad/frontend/templates/404.html]: I thought Django had a default one?

**Comment by terceiro** [comment]
the failure does not happen for you because you don't have the necessary tool installed. you should see a message like this:

```
I: skipping architecture conformance tests (snakefood not available)
```

**Comment by terceiro** [comment]
in practice, you want to add an entry like `('squad.frontend', 'squad.ci'),` to `ALLOWED_MODULE_DEPENDENCIES` in `test/test_architecture.py`

**Comment by terceiro** [comment]
and to run the tests, you want to install `snakefood`

File: squad/frontend/templates/404.html (hunk 3 above)
Comment (mwasilew) [inline squad/frontend/templates/404.html]: If I read the docs properly, it doesn't: https://docs.djangoproject.com/en/1.11/topics/http/views/#the-http404-exception
There was no template when I tested locally so I assume there is no default. Each project needs to create it's own.

**Comment by mwasilew** [comment]
> a few comments. can we also squad all the commits that fix previous commits into a single one?

did you mean squash?

**Comment by terceiro** [comment]
haha yeah I meant squash. I guess my fingers are already too used to typing squad :-)

**Pull request merged**